core/trino-spi/src/main/java/io/trino/spi/connector/ConstraintApplicationResult.java

```diff
@@ -13,25 +13,35 @@
  */
 package io.trino.spi.connector;
 
+import io.trino.spi.expression.ConnectorExpression;
 import io.trino.spi.predicate.TupleDomain;
 
+import java.util.Optional;
+
 import static java.util.Objects.requireNonNull;
 
 public class ConstraintApplicationResult<T>
 {
     private final T handle;
     private final TupleDomain<ColumnHandle> remainingFilter;
     private final boolean precalculateStatistics;
+    private final Optional<ConnectorExpression> remainingExpression;
 
     /**
      * @param precalculateStatistics Indicates whether engine should consider calculating statistics based on the plan before pushdown,
      * as the connector may be unable to provide good table statistics for {@code handle}.
      */
     public ConstraintApplicationResult(T handle, TupleDomain<ColumnHandle> remainingFilter, boolean precalculateStatistics)
+    {
+        this(handle, remainingFilter, precalculateStatistics, Optional.empty());
+    }
+
+    public ConstraintApplicationResult(T handle, TupleDomain<ColumnHandle> remainingFilter, boolean precalculateStatistics, Optional<ConnectorExpression> remainingExpression)
```

Comment:
If I'll do this change, I'll need to change the original constructor at this class to explicitly set all the class's members.
 (instead of just calling this constructor with `remainingExpression = Optional.empty()`).
In addition, `io.trino.metadata.MetadataManager#applyFilter` creates a new `ConstraintApplicationResult`, and if I'll do this change, I'll need to call the correct constructor according to the value of `ConstraintApplicationResult#getRemainingExpression`;
What do you think?